core/components/menu/menu.js

```diff
@@ -471,59 +367,46 @@ Blockly.Menu.prototype.handleMouseEnter_ = function(_e) {
 Blockly.Menu.prototype.handleMouseLeave_ = function(_e) {
   if (this.getElement()) {
     this.blur();
-    this.clearHighlighted();
+    this.setHighlighted(null);
   }
 };
 
 // Keyboard events.
 
 /**
- * Attempts to handle a keyboard event, if the menuitem is enabled, by calling
+ * Attempts to handle a keyboard event, if the menu item is enabled, by calling
  * {@link handleKeyEventInternal}.  Considered protected; should only be used
  * within this package and by subclasses.
- * @param {Event} e Key event to handle.
- * @return {boolean} Whether the key event was handled.
+ * @param {!Event} e Key event to handle.
  * @protected
  */
 Blockly.Menu.prototype.handleKeyEvent = function(e) {
-  if (this.getChildCount() != 0 &&
-      this.handleKeyEventInternal(e)) {
+  if (this.menuItems_.length && this.handleKeyEventInternal(e)) {
     e.preventDefault();
     e.stopPropagation();
-    return true;
   }
-  return false;
 };
 
 /**
  * Attempts to handle a keyboard event; returns true if the event was handled,
- * false otherwise.  If the container is enabled, and a child is highlighted,
- * calls the child menuitem's `handleKeyEvent` method to give the menuitem
- * a chance to handle the event first.
- * @param {Event} e Key event to handle.
+ * false otherwise.
+ * @param {!Event} e Key event to handle.
  * @return {boolean} Whether the event was handled by the container (or one of
  *     its children).
  * @protected
  */
 Blockly.Menu.prototype.handleKeyEventInternal = function(e) {
```

Comment:
Can this method be collapsed into the above ``handleKeyEvent``, with the preventDefault/stopPropagation at the end if it's been handled.

Reply:
Agreed.  I believe that Closure split these up so that they could write cleaner tests against the `handleKeyEventInternal` part.  But we don't have a requirement for that level of testing.